FLIR/conservator/local_dataset.py

```diff
@@ -355,69 +382,61 @@ def download(
         if include_analytics:
             os.makedirs(self.analytics_path, exist_ok=True)
 
-        links = []  # dest, src
-        hashes_required = set()
-        with open(self.index_path) as f:
-            data = json.load(f)
-            for frame in data.get("frames", []):
-                video_metadata = frame.get("videoMetadata", {})
-                video_id = video_metadata.get("videoId", "")
-                frame_index = video_metadata["frameIndex"]
-                dataset_frame_id = frame["datasetFrameId"]
-                if include_eight_bit:
-                    md5 = frame["md5"]
-                    hashes_required.add(md5)
-
-                    name = f"video-{video_id}-frame-{frame_index:06d}-{dataset_frame_id}.jpg"
-                    path = os.path.join(self.data_path, name)
-                    cache_path = self.get_cache_path(md5)
-                    links.append((cache_path, path))
-
-                if include_analytics and ("analyticsMd5" in frame):
-                    md5 = frame["analyticsMd5"]
-                    hashes_required.add(md5)
-
-                    name = f"video-{video_id}-frame-{frame_index:06d}-{dataset_frame_id}.tiff"
-                    path = os.path.join(self.analytics_path, name)
-                    cache_path = self.get_cache_path(md5)
-                    links.append((cache_path, path))
+        # dict stores unique keys in order of insertion. this maps hash -> [links]
```

Comment:
This is true in Python 3.7+, is that the minimum required version in setup.py?

Reply:
In main it has `python_requires=">=3.6"`

Reply:
Should we bump it?

Or replace with `list` of `(hash, links)` and `set` of `hash`.

Reply:
There is also collections.OrderedDict